File: Interview practice/2022/6_12-Bubble_sort.js
/*
https://www.hackerrank.com/challenges/ctci-bubble-sort/problem?h_l=interview&isFullScreen=false&playlist_slugs%5B%5D=interview-preparation-kit&playlist_slugs%5B%5D=sorting
*/

function countSwaps(a) {
    let numSwaps = 0 
    
    for(let y=0; y<a.length; y++){
        for(let i=0; i<a.length-1; i++){
            if(a[i] > a[i+1]){
                let temp = a[i]
                a[i] = a[i+1]
                a[i+1] = temp
                numSwaps++
            }
        }
    }
    
    console.log("Array is sorted in " + numSwaps + " swaps.");
    console.log("First Element: " + a[0]);
    console.log("Last Element: " + a[a.length - 1]);
}

File: Interview practice/2022/6_12-Bubble_sort.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Result{
    public static void countSwaps(List<int> a)
    {
        int numSwap = 0;
        for(int i=0; i<a.Count; i++)
        {
            for(int x=0; x<a.Count-1; x++ )
            {
                if(a[x] > a[x+1])
                {
                    int temp = a[x];
                    a[x] = a[x+1];
                    a[x+1] = temp;
                    numSwap++;
                }
            }
        }

        Console.WriteLine("Array is sorted in " + numSwap + " swaps.");
        Console.WriteLine("First Element: " + a[0]);
        Console.WriteLine("Last Element: " + a[a.Count - 1]);
    }
}